### PR TITLE
 refactor(compiler): move handling of translations to the ConstantPool

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: packages/compiler/src/constant_pool.ts [hunk 1]
```diff
@@ -227,8 +227,12 @@ class KeyVisitor implements o.ExpressionVisitor {
   }
 
   visitLiteralMapExpr(ast: o.LiteralMapExpr, context: object): string {
-    const mapEntry = (entry: o.LiteralMapEntry) =>
-        `${entry.key}:${entry.value.visitExpression(this, context)}`;
+    const mapKey =
+        (entry: o.LiteralMapEntry) => {
+          const quote = entry.quoted ? '"' : '';
+          return `${quote}${entry.key}${quote}`;
+        } const mapEntry = (entry: o.LiteralMapEntry) =>
```
Comment (chuckjaz) [inline packages/compiler/src/constant_pool.ts]: `;` missing

File: packages/compiler/src/constant_pool.ts [hunk 2]
```diff
@@ -213,15 +253,25 @@ export class ConstantPool {
 
   private freshName(): string { return this.uniqueName(CONSTANT_PREFIX); }
 
+  // Closure variables holding messages must be named `MSG_[A-Z0-9]+`,
+  private freshTranslationName(): string { return `MSG_${this.freshName()}`.toUpperCase(); }
```
Comment (chuckjaz) [inline packages/compiler/src/constant_pool.ts]: consider changing this to `return uniqueName('MSG_');` instead which produces `MSG_1`, `MSG_2`, etc. If you really need the `_C` then use `'MSG_C_` as the prefix.

**Review by chuckjaz — APPROVED** [review]

**Comment by vicb** [comment]
Waiting on #22654 to be merged

**Comment by vicb** [comment]
http://test/OCL:190139746:BASE:190139780:1521759187884:d1c7a0f0

**Review by ocombe — APPROVED** [review]

**Pull request closed without merging**